HeartbeatLogging/Tests/Integration/HeartbeatLoggingIntegrationTests.swift

```diff
@@ -13,16 +13,297 @@
 // limitations under the License.
 
 import XCTest
-import HeartbeatLogging
+@testable import HeartbeatLogging
 
 class HeartbeatLoggingIntegrationTests: XCTestCase {
+  // 2021-11-01 @ 00:00:00 (EST)
+  let date = Date(timeIntervalSince1970: 1_635_739_200)
+
   override func setUpWithError() throws {
-    // Put setup code here. This method is called before the
-    // invocation of each test method in the class.
+    try removeUnderlyingHeartbeatStorageContainers()
   }
 
   override func tearDownWithError() throws {
-    // Put teardown code here. This method is called after the
-    // invocation of each test method in the class.
+    try removeUnderlyingHeartbeatStorageContainers()
+  }
+
+  /// This test may flake if it is executed during the transtion from one day to the next.
```

Comment:
Can we use the `dateProvider` to inject the exact same date?

Reply:
We could. The reason I didn't here was to make sure I had at least one integration test that used the public initializer (and defaulted to using `Date.init` as the date provider). 

I realized there was a small potential of a flake in doing so this these tests would run on CI and added the comment to explain why.

WDYT? I'm cool with injecting a `dateProvider` to keep things controlled but am curious about your thoughts on  my thinking above.

Reply:
Reworked in fd19987